Automattic/sensei
# [Refactor] Clean up course outline hooks

### Changes proposed in this Pull Request

* Add module style and border logic to `useSharedModuleStyles`
* Move most logic related to lesson status preview into `status-preview` folder, extract hooks from edit components
* Update naming and add documentation to a few hooks and helpers

### Testing instructions

* Edit a course with modules and lessons, course progress block
* Ensure changing lesson or module preview status works properly and updates everything, including the number of lessons when they are added/removed
